src/enterprise/extension.ts

```diff
@@ -132,20 +132,31 @@ function registerAuthenticationProviders(
   );
 }
 
-async function uninstallGATabnineIfPresent() {
-  // search for the GA extension
-  const tabnine = extensions.getExtension("tabnine.tabnine-vscode");
-  if (tabnine) {
-    // in this case we want to uninstall the GA tabnine extension
+async function uninstallAllOtherExtensionsIfPresent() {
+  return uninstallOtherTabnineIfPresent([
+    "tabnine.tabnine-vscode",
+    "tabnine.tabnine-vscode-enterprise",
+  ]);
+}
+
+async function uninstallOtherTabnineIfPresent(extensionIds: string[]) {
+  const oldExtensions = extensionIds.map((extensionId) =>
+    extensions.getExtension(extensionId)
+  );
+  if (oldExtensions && oldExtensions.length) {
     const uninstall = await confirm(
       "⚠️ You have a conflicting version of Tabnine!",
       "Fix"
     );
     // the user provided consent
     if (uninstall) {
-      await commands.executeCommand(
-        "workbench.extensions.uninstallExtension",
-        "tabnine.tabnine-vscode"
+      await Promise.all(
```

Comment:
did you tested it? promise all will fail if any the  promises is rejected,  now i'm not sure how `workbench.extensions.uninstallExtension` is behaves when a non existing extension name  is sent, but id change the `Promise.all` to `Promise.allSettled` and check if any is fulfilled